nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/ConsumeAMQP.java

```diff
@@ -194,6 +194,13 @@ protected synchronized AMQPConsumer createAMQPWorker(final ProcessContext contex
 
             return amqpConsumer;
         } catch (final IOException ioe) {
+            try {
+                connection.close();
+                getLogger().warn("Closed connection at port " + connection.getPort());
+            } catch (final IOException ioeClose) {
+                throw new ProcessException("Failed to close connection at port " + connection.getPort());
```

Comment:
You're at this point because of an io exception that already occurred.  Here you're closing a connection as good practice but the fact that it too might not go well isn't useful/not something you could do anything about.  As such I'd closeQuietly meaning do not rethrow the exception and do not log even at a warn level.  If you want you could keep the logging part and only as a debug thing.

Reply:
would it be a good idea to swallow the ioeClose exception by means of a warn-log?
```java
        } catch (final IOException ioe) {
            try {
                connection.close();
            } catch (final IOException ioeClose) {
                getLogger().warn("Failed to close connection at port " + connection.getPort(), ioeClose);
            }

            throw new ProcessException("Failed to connect to AMQP Broker", ioe);
```